src/librustdoc/core.rs

```diff
@@ -229,7 +229,8 @@ pub(crate) fn create_config(
         if proc_macro_crate { vec![CrateType::ProcMacro] } else { vec![CrateType::Rlib] };
     let resolve_doc_links =
         if *document_private { ResolveDocLinks::All } else { ResolveDocLinks::Exported };
-    let test = scrape_examples_options.map(|opts| opts.scrape_tests).unwrap_or(false);
+    let test =
+        scrape_examples_options.map(|opts| opts.scrape_tests).unwrap_or(false) || *document_tests;
```

Comment:
> > > I think it's a good idea. Just one thing missing: an entry in the rustdoc book about the command line flag and that it needs to be used with `--document-private-items` and likely with `--cfg test` as well.
> > 
> > 
> > It would be quite easy for rustdoc to automaticly set the `test` cfg when `--document-tests` is passed, in the same vain as done with the `doc` cfg.
> > https://github.com/rust-lang/rust/blob/82d17a4db3a010caa947d49245de63a7ac14accf/src/librustdoc/core.rs#L206-L207
> 
> Yes it's easy to do. The question is more: "should we do it?".

In this implementation, ' --cfg test` is already passed automatically here.
It may not be the cleanest way to pass it.

Having it passed automatically is convenient, I think it would be a bit counter-intuitive if the option "document-tests" did not actually document tests except if an additional flag is passed.

Reply:
I think it's an important thing to discuss as it might not be obvious for users. I would personally pass `--cfg test` as I wouldn't expect an option to modify the `cfg`s.

Reply:
possible relevant MCP https://github.com/rust-lang/compiler-team/issues/785

Reply:
FYI, https://github.com/rust-lang/compiler-team/issues/785 has now entered FCP.